subsys/bluetooth/audio/bap_broadcast_assistant.c

```diff
@@ -108,6 +109,17 @@ static int16_t lookup_index_by_handle(uint16_t handle)
 	return -1;
 }
 
+static struct bap_broadcast_assistant_instance *inst_by_conn(struct bt_conn *conn)
+{
+	struct bap_broadcast_assistant_instance *inst = &broadcast_assistants[bt_conn_index(conn)];
```

Comment:
it has already been guard before calling inst_by_conn() in the cases which conn could be NULL

Reply:
As long as this is a `static` function I'm OK with not having the `conn == NULL` check, as we can ensure that where we call this. 

That being said, `notify_handler` does call this without a check, and IIRC that function can be called from GATT with `conn == NULL` if there is a disconnect on an unbonded device

It may be better to simply add the check in this function to make sure